### ISSUE TITLE
"Block user" feature is never used as intended and is not well designed

### ISSUE THREAD

**Issue by Voileexperiments** [Codewars/codewars.com#1365]
So, if I've learnt my CW history lessons properly, the current block user feature was added because of #623 and #787.

Currently these things happen if a user (A) has blocked another user (B):

- A cannot see B's comment, they're masked and will say: `This comment can not be viewed.`
- B cannot see A's profile
- B cannot leave comments on A's profile (at least on the front-end)
- B cannot edit A's kata
- ...

The problem: it was added due to *one* user being annoying, and this has never happened again in the future. At least for the whole year I'm on CW, I've never seen any actual occurrences of any flame war happening, so it's solving a "problem" that doesn't exist in the first place.

(Note that even things like #1198 and #1210 had happened *just once* before, and it was resolved by *admin action*, not block user.)

However, what I *have* seen multiple times is newbies blocking users telling they're wrong, and kata authors blocking users to stop these users posting issues on their katas (I'm looking at you, @SteffenVogel79). The latter is not normal usage of this feature. It's an abuse.

Also, block users feature should not *hinder normal site usage*. CW does not belong to any user, so there's no reasons being blocked by a user makes me unable to perform certain *site*-related actions. This is not Twitter or Facebook, we don't have user-specific timelines, just public katas that everyone can see and comment on.

---

Proposal:

1. Either remove all restrictions on those `B cannot do this to A` and just leave the first one apart...
2. Or just remove this feature, it doesn't solve anything practically and track record proves that it's causing hindrances instead. It wasn't even added with much thought in the first place.

**Comment by SteffenVogel79** [comment]
Voile, you are creating issues only because you think a kata have to be in your way of thinking.
Your postings are annoying and I don't like your revenge bad votings.
That is the reason, I blocked you! I do not want to discuss or read from you! So, the blocking feature is the best and most worthy feature on codewars.
Have a nice time on codewars, but please without disturbing and annoying me.

**Comment by Blind4Basics** [comment]
Though he is right about one thing: shit posting authors that can block power users so that they cannot post issues on their wrongly done katas IS a REAL PROBLEM.

**Comment by Voileexperiments** [comment]
@SteffenVogel79 What you're saying is completely wrong and off the basis of facts.

For the record, you blocked me when I posted issues at your "Construct a Car" katas:

https://www.codewars.com/kata/constructing-a-car-number-2-driving
https://www.codewars.com/kata/constructing-a-car-number-3-on-board-computer

The thing is, just like most of the OO katas on CW, these katas have *serious lack of specs*:

- 2: See the comments: did you see that at least 3 people are asking about what some of the tests are about? I was posting an issue about 2 of the things that I encountered as I'm attempting the kata, *and I haven't even gone through the whole thing by that point*. There are even more hidden mines.
- 3: It's basically 2 but lots, lots more of these mines. The consumption by distance metric are the worst: only in actual tests, zero feedback, zero information as to how the tests are set up, and in fact even the calculation of it is vague. Basically the tests are shit.
- Also, lots, *lots*, **lots** of tests are in the actual tests only and they test behaviours which is not stated anywhere. If we have to comtemplate

I mean, seriously, both you and @g964 needs to get your heads straight and admit the fact that **your kata can be unclear/have issues and you have to work with the users to resolve them, not just f\*\*king block/shoot down the reporters and pretend they don't exist**. Has anyone felt good about how g964 handled the inquiries in his kata comments? Pretty much none. In fact it's so annoying it's been vented out on various occasions; we just don't speak out in the kata comments.

And I have to say, you're even worse than him. Shooting down issues by abusing a site feature? This is bad and you should feel bad. You're not even trying to be cooperative to your kata users (there were lots of questions in your katas about the specs and you answered none).

(Shots fired because, seriously, lots of people have been thinking along the same line. I think *someone* has to speak out, and it's gonna be me today.)

**Comment by kodejuice** [comment]
:)

**Comment by FArekkusu** [comment]
Bumping the issue. This "feature" is abused more than used for valid reasons. Considering the fact, that CW is "calm" and more-or-less friendly in the whole, there's virtually no reason why users would need to block other users. On the other hand, oversensitive kata-creators (let's be fair, they are the only ones doing this) like to block users whose views (especially on kata issues) they do not share. And this is a **far more serious** problem than a possibility that once in a few years some troll will start spamming messages and targetting users (this also doesn't help with the targetted downvoting bullshit).

**Comment by Voileexperiments** [comment]
Is this going to be fixed? Because apparently SteffenVogel is just blocking everyone who raises a issue on his katas. This is literally the worst abuse of this "feature" (if it's even a "feature" in the first place).

**Comment by FArekkusu** [comment]
Another problem is that it *may* fix the issue if user A is exclussively targetting user B, but it changes nothing if troll is spamming everywhere. Obviously, all of the active users won't just go and block him, and even if they did, the endless flow of `"this message is not shown to you"`'s would still appear on the dashboard.

**Comment by kazk** [comment]
I've been avoiding this issue because I don't really know how to "fix" and I didn't know what to say. I've already spent more time than I wanted to so here's my initial thoughts.

---

@Voileexperiments 	

> At least for the whole year I'm on CW, I've never seen any actual occurrences of any flame war happening, so it's solving a "problem" that doesn't exist in the first place.

I see you arguing with other users (mostly new users) about best practices very often. I understand the frustration, but I'd appreciate if you toned down. They're usually just lacking knowledge and/or experience and simply need more time to gain that. Words from some random person won't be enough anyways.

Also, sometimes you sound too harsh. Again, I understand the frustration of seeing similar issues happening over and over, but usually those are different users.
I _personally_ don't mind the directness since I believe you have good intentions, but many will misunderstand you. They will react defensively and the communication will be highly inefficient. So I'd appreciate if you can pause and think about how others will feel before posting your comments.

---

@SteffenVogel79 

Can you help me understand the problem from your point of view? You seem to have blocked many users. Can you elaborate on how they annoyed you?

I know @Voileexperiments sometimes sounds too harsh, but issues opened for the kata mentioned above seems fine. If there was other issues that triggered blocking him, please let me know so I have a better understanding.

- [Constructing a car #2 - Driving](https://www.codewars.com/kata/constructing-a-car-number-2-driving/): `2279` code submissions and completed `49` times.
- [Constructing a car #3 - On-Board Computer](https://www.codewars.com/kata/constructing-a-car-number-3-on-board-computer): `471` submissions and completed `8` times.

Doesn't this show many are struggling to understand the requirements? Or they had to submit in order to find the hidden requirements? Do you think descriptions _can_ be improved?

It's your kata, but it's for the Codewars community to solve and to learn from. So I'd like you to **at least consider whenever someone opens issues rather than forcing them to stop by blocking.** They're usually trying to help improving it or having problems understanding it.
If you disagree with them, explaining the reasons behind your decision will be much more constructive. This can also prevent someone else from raising the same issue again.

Like I wrote above, I only looked at the mentioned kata. So let me know if I've misunderstood the problem.

> So, the blocking feature is the best and most worthy feature on codewars.

This is very depressing :(

**Comment by g964** [comment]
I have never blocked anyone but I think this should be kept if used wisely. "Respect and courtesy" must be CW's motto.

**Comment by g964** [comment]
I have never blocked anyone but I think this should be kept if used wisely. "Respect and courtesy" must be CW's motto.

**Comment by Voileexperiments** [comment]
@kazk I do not consider my tone "too harsh" for the situation in which I choose my tone; I choose my tone that I feel reflects what the other side deserves, which includes the tone of the original message and the other side's reputation. *Being respected is a privilege that is earned by being, well, respectable; it is never an universal right in the first place*. Yes, I do admit sometimes I get salty like everyone does, but I do adjust afterwards.

What is not mentioned here is that most of the times I'm very strongly worded because the other side starts the discussion with a condescending tone, especially prevalent with "I can't pass the test" inquiries-as-issues and UUN (Unreadable, Unmaintainable, Not for production) arguments: what do you expect to do then? Newbies *buy into these arguments*; and sometimes *the entire discussion thread is wrong*. If you don't debunk them, you lose by forfeiting the argument. If you're not strongly worded enough, you lose because you look like a wimp. I know people don't like to listen to strong worded sentences because it hurts their (sometimes blistering) ego, but I choose to do it anyways because while every choice lead to losing, it is at least the best for my mental health. I hope you can understand my situation.

(Also, humbleness goes to both sides too: if one is inexperience, they should've been sure to be extra humble and not make very strong claims because they're more likely to be incorrect. This is like, I dunno, human interaction 101? If someone's bold enough to make outrageous claims about wrong stuff despite the lack of abilities and doesn't back up at all, I feel I have all the rights to retaliate.)

---

I am *very strongly worded* about this issue because it is, to be honest, very appalling. It takes effort and time to actually pinpoint and write the issues; after all I can just solve the kata in the quickest way, do nothing about the obvious problems and let everyone after me suffer from the same thing. But then all the beta katas will be approved with tons of issues, and in the end kata quality becomes abyssmal, CW quality is dragged down as a result and everybody loses.

I know nobody wants to have issues in their katas, especially the bogus ones. I understand that. But blocking users to prevent them from posting issues because the author *believes* that their issues are "not a issue" is not just ridiculous, but completely disrespectful of any author-user interactions. The author might as well live in an ivory tower where they believe that their works are "perfect", and anyone who dares to point out otherwise would be banished. Now *that*'s the real WTF right there.

(In addition, for almost all the katas with issues I can just edit them and change everything to make them make sense again, which would've saved all the troubles and avoided any needs to deal with potentially hostile kata authors. I choose not to because *I still respect the will of the kata author*. It's their katas, and they get to decide what the requirement is. By committing the act to *outright blocking users who post issues*, the kata author is basically saying they do not want to be respected. So why should I?)

Oh, by the way, yes, those two katas still have issues with completely unspecified requirements and wrong requirements.

---

Also, let's point out *more* problems with how the current "block" function is implemented:

- As mentioned above, it is supposed to be a restriction on user interaction but it's interfering with site functions. Site function related restrictions should've been handled by admins (like a certain `no one` that has manipulated the beta system by somewhat). If I can censor people I don't like by using the block function like this, it's abuse.
- It is not just visible, it's wide in the open: I can easily see a kata author has blocked me from the comment section of their katas. In fact it's even worse: #1357 is not patched so anyone can instantly see all users that have blocked them.

I recommend reading [this](https://alexwlchan.net/2018/09/assume-worst-intent/) as to how to design these user interaction related features that prevents abuse.

**Comment by kazk** [comment]
@Voileexperiments

You're assuming that they can recognize their lack of ability.
They _think_ they know what they're talking about because that's all they know. If they only know one way of doing something, they won't be able to think about trade-offs.

UNN comments are often reactions to seeing something different from what they value and not being able to accept it. They're in denial. They have the right to express that and they should be able to have civil discussions. It's part of a learning process and there's no need to put them down.

"I can't pass the test" is someone being desperate and not knowing what to do. We should tell them to label their comment as questions instead and help them.

> Newbies buy into these arguments; and sometimes the entire discussion thread is wrong. If you don't debunk them, you lose by forfeiting the argument. If you're not strongly worded enough, you lose because you look like a wimp.

I honestly don't understand this.
Do you argue with little kids and yell at them for not knowing better?
I mean, they might be wrong, but that's not really their fault. They just need to learn.

---

> It takes effort and time to actually pinpoint and write the issues; after all I can just solve the kata in the quickest way, do nothing about the obvious problems and let everyone after me suffer from the same thing.
>
> I choose not to because _I still respect the will of the kata author_. It's their katas, and they get to decide what the requirement is.

I know you have good intentions.
You clearly care about Codewars and have done a lot for the community.
Unfortunately, many users don't think so because they've misunderstood you and they're intimidated by you. A new user might see your comment on the dashboard and assume you're abusive user putting others down.
You might not care what others think of you, but I think that's sad and that's why I mentioned it in the first comment.

---

I agree blocking shouldn't be used this way, but I also understand some authors want to do it their way.
For the next version of Codewars, I was thinking of having a few separate pool of kata:

- Official Kata Listing
  - Require certain quality to be listed here
  - Authors agree to adhere to Codewars standards and are not allowed to personalize the content
- Personal Kata Listing
  - Authors are free to do whatever they like and it'll be under their namespace
- Beginners Kata Listing
  - Kata for really basic stuff to learn languages

---

> Also, let's point out more problems with how the current "block" function is implemented:

I agree it's not implemented well. Thanks for the link, it was a great read.

**Comment by kazk** [comment]
@SteffenVogel79 probably muted this issue.
I just checked if any other author does this and he is the only one that blocked multiple active power users. Maybe we can keep it as is for now.

**Comment by Voileexperiments** [comment]
PSA: I've just been blocked by JohanWiltink too. He's probably fed up with my issues on his kata.

Relying on users to use the `block user` feature responsibly when we all know it can slience users you don't like on your katas is just not gonna work. Are we still gonna repeat the same history again?

**Comment by SteffenVogel79** [comment]
Strange. I did not block Johan, and he did not block me.
Bot we blocked YOU both! Maybe the problem is you!

**Comment by Voileexperiments** [comment]
What do you mean? Why don't you list out the 1000 users you've blocked? ;-)

You're the biggest prick on CW living on the highest ivory tower, and every power user knows it. I'm surprised you have the audacity to *still* comment on this issue.

**Comment by SteffenVogel79** [comment]
It must be so hard for you, that there are users, who don't want to read your stupid comments... And they can block you... sorry, it must be very hard.

There are many power users, with whom I have discussed hard and we are still ok with each other.
But freaks like you.... no thanks.

Bye, voile. The blocking-feature ist the best feature here ever!

**Comment by Voileexperiments** [comment]
What? Nobody is trying to win him over.

I think you're getting the point wrong: he has been doing this for *years* and has basically refused any kind of communication (just look at this issue). If anything most PUs *do not* want to deal with him at all.

This issue is opened to advocate kazk to remove most of the (unnecessary and poorly designed) power associated with block feature to dethrone him from the ivory tower he's on right now.

**Comment by plasticbottle** [comment]
Well then don't deal with him... ignoring each other seems to be the simpler way here, it doesn't look like there is any chance of reconciliation, so maybe mutually ignoring each other works better rather than try to change the whole website because of one single particular dispute between two users. But that's just my $0.02.

**Comment by plasticbottle** [comment]
Yes, I have read through the issues, although I am not such an active or great user, and I'm nowhere near as experienced with coding as you, I do find the progression of this website interesting. I do not mean/did not mean to attack you personally or something, but I just think that this issue is completely pointless and just wanted to voice my opinion about that. I hope that's fine with you. 

If someone blocks too many people needlessly, it will become their own problem too, you know, one day they will have nobody to interact with. Also, I don't think that calling people vulgarities will help your cause nonetheless, it's kind of hard to take it seriously at that point.

I don't think that a community site with no block feature will work, sometimes people just don't get on, and there needs to be a way for such people to stop the other one from interacting with them.

**Comment by error256** [comment]
The thing is that when a user blocks everyone, there's no one left to raise issues on their katas, so there're lots of permanently unfixable katas around.

**Comment by plasticbottle** [comment]
> The thing is that when a user blocks everyone, there's no one left to raise issues on their katas, so there're lots of permanently unfixable katas around.

Maybe a (reasonable) limit on the number of users that you can block would help. But not removing the feature altogether...

**Comment by SteffenVogel79** [comment]
Neithor Johan nor me blocks everyone! Only freaks like Voile are blocked! 
The number of users blocked by me is not so high as Voile said.
He is really my important blocking!

**Comment by Voileexperiments** [comment]
> Maybe a (reasonable) limit on the number of users that you can block would help. But not removing the feature altogether...

Why? Katas do not belong to authors. They belong to the site. By submitting your kata to CW you already forfeit your right to the original content you've written: they become CW material. This is how every site featuring user-created content works.

So no, blocking users causing a disruption in site function does not make sense, because that would imply the kata author is "special", in which they are not. Kata author do not, and should not have the privilege to block users to access to a kata's specific features.

Also, there is still the issue that block user feature is prone to abuse, and *has* been abused (by multiple users even). Are you going to address it? If not then this issue is not for you.

**Comment by plasticbottle** [comment]
> If not then this issue is not for you.

Thankfully it is not up to you to decide what is for who :)

I see you are often berating other users (sometimes with vulgarities). If we go by standards of how other sites work, you'd be long excluded by now. 

If you have not considered the fact that people do not want to be berated by others, then you probably should. It's the main issue that you are having (and causing to yourself).

**Comment by Voileexperiments** [comment]
In any case, are you contributing to the discussion? Or are you just trying to invalidate my personality?

Looks like both you and SteffenVogel is doing the latter. I thought you would be better than this ;-)

**Comment by plasticbottle** [comment]
Yes, I am, I'm giving you a neutral point of view. It's not about invalidating your personality, it's about the fact that you are berating other users and then posting (pointless) issues that they are blocking you as a response.

**Comment by Voileexperiments** [comment]
Your "neutral" point of view is funny, because in a neutral point of view I do not enter the equation, nor SteffenVogel.

A neutral point of view is this:

- Blocking users currently disrupts normal site functions
- Kata do not *belong* to kata authors, so kata author should not be able to rig their katas in whatever means

Hence I already mentioned my suggested course of action above:

- Block user do not need to be removed, but it should be revised
- It should *definitely* not block users from interacting with katas, because katas do not belong to kata authors. If there are abuse (which there isn't btw), they should be resolved by admins, not some dumb block functions
- However blocking users should hide their comments. That's why blocking are supposed to be in the first place

And when will you learn to not talk about individual users? You're really "neutral" there, you know ;-) Maybe you should take your effort to post on #1646 instead.

**Comment by plasticbottle** [comment]
> And when will you learn to not talk about individual users?
> (I'm looking at you, @SteffenVogel79)

Your opening comment is literally about picking out an individual user :) 

> Block user do not need to be removed, but it should be revised

Agreed, the site does not seem to have big issues in general, perhaps a limit to how many users can be permanently blocked could work, and perhaps there could also be a feature to block someone temporarily.

> It should definitely not block users from interacting with katas, because katas do not belong to kata authors

Do not agree, people should be able to have _complete_ peace of mind from abusive users.

**Comment by kazk** [comment]
@SteffenVogel79 did you read [my comment](https://github.com/Codewars/codewars.com/issues/1365#issuecomment-431741182)? Can you reply to that instead of calling names?

---

I agree it needs improvement, but this is low priority because I already have too many problems to fix and this doesn't seem too bad:

- Number of kata authors that blocked at least one active power user: 15
  - 14 of them blocked one or two
    - 12 of them blocked one
    - 2 of them blocked two
  - 8 of them authored 3 or less kata
- Number of kata authors that blocked at least one active user: 18
  - 17 of them blocked one or two
    - 14 of them blocked one
    - 3 of them blocked two
  - 9 of them authored 3 or less kata

As far as I can tell, @Voileexperiments (blocked by many) and @SteffenVogel79 (blocked many) are outliers.
I can't look up those blocks temporarily so I don't know about that.

**Comment by CliffStamp** [comment]
The main issue is that there isn't a separation between

- general feedback, commentary
- raising issues

It is reasonable for a user to be able to prevent the first, completely unreasonable for the second. 

If they are not uncoupled then a solution for users who do the second would be they lose the right to do the first. However what is happening in some cases is that one user feels another one is raising issues which are not actually issues and they are blocking them to prevent that. That is a more complicated issue because someone has to judge if the issues are valid to settle the conflict.

**Comment by Voileexperiments** [comment]
@kazk Well, #1357 is not fixed yet, so...

I'm not sure if even my case is "too many" compared to what SteffenVogel is doing. There are users with 2 honor, there are many inactive users that haven't even posted comments, and after all there are just... 3 active users? At least compare that to what PUs have been mentioning the in gitter chat over the year about being blocked by him, that's certainly much lower in numbers.

Also, I should point out that @plasticbottle have also blocked me. I think I understand now why he insists on talking about *me* instead of the issue.  (It's also why disclosure of interest is a good thing!) For some reasons the blockers have been acting are low-quality trolls in this issue instead of, say, contributing to the issue.

---

I do agree about

> If they are not uncoupled then a solution for users who do the second would be they lose the right to do the first.

though. But then this is again material for #1646, and I think we need more of kazk's feedback.

**Comment by plasticbottle** [comment]
> Also, I should point out that @plasticbottle have also blocked me. I think I understand now why he insists on talking about _me_ instead of the issue

I did that as a preventative measure, since I do not want to spend much time arguing with users that like to do that all the time. All I can say is reiterate what has already been said in #805:

> If a blocked user feels like they can't express themselves because they were blocked - maybe they should reconsider their actions that got them blocked in the first place.

**Comment by Voileexperiments** [comment]
> I think they'd expect that the blocked user can't contact them anymore anyhow.

CW is not Facebook, and the discourse is not owned by anyone ;-) I don't care if the block function make you unable to see my comments (after all, everyone is free to ignore other's comments), but if I cannot comment on some of the *entire* discourse because I was blocked by an user, then I consider it a breach of my rights as I have the right to comment on *every* discourse, unless I had done something that forbids me to comment on *the entire site*. This has happened before, in fact twice (#1198, #1210) ;-)

The same goes to katas. Kata *does not* belong to kata author, so they do not have the right to arbitrarily deny an user's functionality on their katas. By tomorrow you might as well claim that by blocking me I'm not able to *solve* any of your katas.

In fact I think only two person here would think so ;-)

> All I can say is reiterate what has already been said in #805

And in the same thread jhoffner also said "I was reluctant to add it in the first place". When you quote someone, please quote it in full and not just cherry-pick the part that flavours you ;-)

Everyone knows that why and how the block feature was implemented (see this issue's OP), and how much of a rush it was. Blocking users also blocks the entire discourse is *not* part of #623, and 2 of the 3 proponents there have blocked me anyway (yes, I figured out dinglemouse blocked me recently too. JohanWiltink has lifted the block though).

So what I'm saying here is: essentially jhoffner just listened to **3** particular outlier users and hastily (in 2 days) implemented a feature due to a conflict between one of them and another user; in which this feature is being abused by also the same outlier users. There were no feedback asked from any other users. In this ground I cannot claim at all that this function is well-designed at all, or is implemented in the benefit of most CW users. No, not at all.

(Also it's good to know that SteffenVogel has already behaved like that back in #623 was made. Really eye-opening!)

---

Also, let's get this straight: I perfectly know you don't like how I talk. But I don't care :trollface: Keep telling me this is not doing anything, so you might as well do other things instead.

**Comment by plasticbottle** [comment]
>  Hence the title of this issue, and my strong request that it should be removed before it is certain that most users actually want this function.

Good. I respect your opinion and your view. I am here to say the opposite however. People should be able to have complete peace of mind from users that they find abusive/obnoxious. As much as you have a right to express your opinions, everyone has the right to feel comfortable on the website too.

So what do you _really_ want to achieve? Troll everyone on this website until everyone else but you gets frustrated and leaves? :) 

> (see this issue's OP)

There you pretty much picked out a user and made a salty thread that you can't harrass him anymore. Why don't you try instead, to find out why he blocked you, and try to say sorry/apologize and reconcile, instead of calling him out publicly on a forum, and trying to force the site admins to lift the block? 

> I perfectly know you don't like how I talk.

If you are such a leading figure in being blocked, then apparently I'm not alone. Perhaps community advices shouldn't be taken from you...

>  But I don't care :trollface:

This response entirely reflects your overall attitude.

**Comment by SteffenVogel79** [comment]
The fact, that Voile called me a prick is the best sign, that it is the best decision to block him!

**Comment by Voileexperiments** [comment]
> I am here to say the opposite however. People should be able to have complete peace of mind from users that they find abusive/obnoxious.

And CW should be ensuring that kata authors are fulfilling their responsibility and doing their jobs properly, which SteffenVogel has done none of them. He's just abusing this feature so he can continue not fulfilling any responsibilities as a kata author and shoot down every single issue in his kata.

If it is just used to slience particular user, I don't mind (like I don't complain about how KenKamau blocks people temporarily). But if it is used to circumvent the kata process, this is an abuse.

And CW > people, so your opinion is invalid.

If you want to argue, don't argue *about* me. You're still spending an entire thread commenting about me. I thought you should've learnt that already ;-)

**Comment by plasticbottle** [comment]
> And CW > people, so your opinion is invalid.

Umm... what? There is no CW without a userbase. If it becomes possible to keep trolling a person and the person has zero tools to stop that as per your suggestion, it means the only thing a user can do is basically delete his account and leave the website and thereby letting the troll win. And that is not cool :(

The current system seems reasonable. People can block each other, instead of going down on big flame-wars. Additionally, the current system forces people to act respectfully to each other, since there is now a real deterrent from trolling authors. 

> If you want to argue, don't argue about me. 

I do not really care about you, but I did not like how a user in OP and ever since OP has been getting picked on. And the abusive language got me to put my $0.02 into this thread.

> And CW should be ensuring that kata authors are fulfilling their responsibility 

I saw a pretty good idea from the site admin in another thread for the future, I'll link it for you:
https://github.com/Codewars/codewars.com/issues/1626#issuecomment-438400435

Perhaps we simply need to be patient ;)

**Comment by SteffenVogel79** [comment]
Last week I solved an issue, that was correct.
But I close issues like "Your description is not in the way I like" or "My code does not work. Your tests must be wrong!"

And I blocked Blind4Basic. This was really a fault and I unblocked him. Sorry, Blind4Basic, if you read this.

But again and again, it shows, that it is the best feature to block freaks like Voile! Really!

**Comment by FiveCrayFish973** [comment]
I kinda agree with @SteffenVogel79. Sometimes there are users that need to be blocked, I for one have not blocked anyone since I failed to see anything wrong with anyone (highest in my list is @FArekkusu and I don't hate him enough to bother blocking). But if some users want some peace of mind, they should be allowed to block a user. @Voileexperiments on pointing out issues : just give it a down-vote and rank it, and post in gitter channel that there is an issue on this Kata and let someone else post it (neither of them have blocked Blind4Basics, ZED and a ton of others).

**Comment by DonaldKellett** [comment]
So after [one particular user](https://www.codewars.com/users/lbvf50mobile) threw a fit on [a hastily authored "Kata"](https://www.codewars.com/kata/just-minus-one-from-a-huge-number-that-represented-as-a-string/discuss/javascript) and [this repo](https://github.com/Codewars/codewars.com/issues/1866), he proceeded to block me on Codewars solely based on a (perhaps out-of-place, I'll have to admit) comment I left on the [Codewars Gitter channel](https://gitter.im/Codewars/codewars.com) on his last reply to #1866 . As this particular case once again demonstrates, the block feature is commonly abused by *irresponsible* Kata authors who don't give a s\*\*\* about the Codewars community in order to push their s\*\*\*ty Kata out of Beta.

@kazk @jhoffner Can we just agree to remove this "block" function once and for all? As @Voileexperiments stressed multiple times before:

- It was *hastily* added to Codewars based on just **one particular incident** of some egotistic maniac (who I won't name here to protect his identity from those who do not already know him) causing shenanigans in the Discourse section of multiple CW Kata who requested this feature *himself* to prevent *others* from refuting his absurd actions
- It has brought *nothing* but harm to the site. If a particular Codewarrior is found to exhibit disruptive behavior on Codewars by leaving inappropriate comments or otherwise then *an admin can always step in and ban the user from continued engagement of Codewars' services*. Instead, Codewars has opted to provide a "Block" button which gives the perfect excuse for *snowflakes* unable to handle comments from respected Codewarriors pointing out **actual issues** on their Kata to "ignore the elephant in the room", so to speak, by blocking them and continue on with their destructive actions
- *It doesn't actually work*! As #1840 has demonstrated, experienced Codewarriors can easily work around the "block" by fiddling with various endpoints creating issues that the Kata author can neither view nor resolve

**Comment by Harlot** [comment]
I'm going to have to pipe in and comment that the biggest issue with the "Block" feature is it does not understand the problem that it is trying to solve... and in the history of the Internet has never done anything but escalate the scenario.

I've gotten into a few heated interactions with people back when I was still checking stuff out on Codewars (I've since then decided to take a break for a bit, to see if that has me more level headed when I return)... but even when talking about how the to solve me getting into these interactions, I'd have not suggested this.

Essentially, there are a number of issues from Dunning-Kruger Effect, Tribalism, Confirmation Bias and a result in blocking just making stuff worse.

IIRC, I mentioned that such a feature would only make matters worse prior to me taking a little self-imposed vacation last year.

That being said, the closest thing I can give to a solution is "have moderators/staff be trained in psychology sufficiently"--which is a fucking shitty solution that is just awful.

**Comment by Voileexperiments** [comment]
Since then there are apparently at least several cases where *new users*, after publishing the first beta kata and being pointed out their beta kata do not constitute acceptable quality, begin *blocking* power users that are pointing the above out:

https://www.codewars.com/kata/student-grades-what-grade-did-the-student-achieve
https://www.codewars.com/kata/is-fred-a-good-golfer-or-not

...how are the new users knowing this feature? Is there are group of people in behind who are actively trying to sabotage the beta process and undermine the power users or something? Because otherwise it's difficult to explain how new users magically know this feature undocumented on anywhere else besides here. This is honestly ridiculous and we're now resorting to dirty tactics, apparently?

**Comment by FArekkusu** [comment]
The situation described by Voile aboved has just happened again [here](https://www.codewars.com/kata/5e1f2aa71e4f11000fe308f2). Some new user authored a terrible kata, and after a few hours of being criticized that it provides zero value to CW he unpublished it and blocked everybody while also calling PUs "trolls".

I understand that removing a feature is not an easy process, especially on CW, but can the `Block` button at least be removed from the user profile HTML file, and all the previous blocks invalidated?

___

This is also another example why the current 300 honor requirement to become an author is ridiculously low :(

**Comment by Blind4Basics** [comment]
seems he unblocked us (at least, I can see his profile again, for now)

**Comment by Voileexperiments** [comment]
It seems that for the several cases where new kata author blocks power users after being raised issues by by said power users, after being pointed out what they're doing is absurd they unblock the power users after a short while. So I do think they're aware of what they're doing (and how this feature works), which is pretty suspicious IMO.
